### PR TITLE
complete include refactoring

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- remove `simulation_defines.hpp`
- add `param.hpp` to get access to param files for a subgroup of the code
- fix missing includes
- remove `extentionParam.loader` and `extensionUnitless.loader`

This refactoring is again resolving a bunch of cyclic includes and missing includes.

### CODE REVIEW

File: include/picongpu/fields/absorber/param.hpp
Comment (PrometheusPi) [inline include/picongpu/fields/absorber/param.hpp:1]: What is this change?

File: include/picongpu/fields/incidentField/param.hpp
Comment (PrometheusPi) [inline include/picongpu/fields/incidentField/param.hpp:1]: again, what is this change?

File: include/picongpu/logging.hpp [hunk 1]
```diff
@@ -19,21 +19,12 @@
...
-
-
 namespace picongpu
 {
-    using namespace pmacc;
-
     // short name for access verbose types of picongpu
     using picLog = PIConGPUVerbose;
 
+    using pmacc::log;
```
Comment (PrometheusPi) [inline include/picongpu/logging.hpp:29]: why does this single line solves all issues with the removed includes above?

File: include/picongpu/particles/DeviceHeap.hpp
Comment (PrometheusPi) [inline include/picongpu/particles/DeviceHeap.hpp:1]: Is this change intended? 
Looks like a new file. Why was it needed?

File: include/picongpu/particles/InitFunctors.hpp [hunk 2]
```diff
@@ -108,7 +109,26 @@ namespace picongpu
 
                 DensityFunctor densityFunctor(currentStep, idProvider->getDeviceGenerator());
                 PositionFunctor positionFunctor(currentStep, idProvider->getDeviceGenerator());
-                speciesPtr->initDensityProfile(densityFunctor, positionFunctor, currentStep);
+
+                log<picLog::SIMULATION_STATE>("initialize density profile for species %1%") % FrameType::getName();
+
```
Comment (PrometheusPi) [inline include/picongpu/particles/InitFunctors.hpp:114]: This looks like new functionality nut just updating of `initDensityProfile`

File: include/picongpu/particles/Manipulate.def
Comment (PrometheusPi) [inline include/picongpu/particles/Manipulate.def:1]: fullly new file

Comment (PrometheusPi) [reply include/picongpu/particles/Manipulate.def:1]: Okay its from `Manipulate.hpp` - fine

File: include/picongpu/particles/Particles.tpp [hunk 3]
```diff
@@ -376,37 +376,6 @@ namespace picongpu
         ParticlesBaseType::template shiftParticles(mapperFactory, onlyProcessMustShiftSupercells);
     }
 
-    template<typename T_Name, typename T_Flags, typename T_Attributes>
```
Comment (PrometheusPi) [inline include/picongpu/particles/Particles.tpp:379]: Why could this be removed?

File: include/picongpu/particles/param.hpp [hunk 4]
```diff
@@ -1,5 +1,4 @@
-/* Copyright 2013-2023 Axel Huebl, Heiko Burau, Rene Widera, Felix Schmitt,
- *                     Marco Garten, Finn-Ole Carstens
+/* Copyright 2013-2023 Axel Huebl, Rene Widera, Sergei Bastrakov, Klaus Steiniger
```
Comment (PrometheusPi) [inline include/picongpu/particles/param.hpp]: authors disappeared from the license header

File: include/picongpu/plugins/output/param.hpp [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2013-2023 Rene Widera
+/* Copyright 2013-2023 Axel Huebl, Rene Widera, Sergei Bastrakov, Klaus Steiniger
```
Comment (PrometheusPi) [inline include/picongpu/plugins/output/param.hpp]: authors added

File: include/picongpu/plugins/particleCalorimeter/param.hpp [hunk 6]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2013-2023 Rene Widera
+/* Copyright 2013-2023 Axel Huebl, Rene Widera, Sergei Bastrakov, Klaus Steiniger
```
Comment (PrometheusPi) [inline include/picongpu/plugins/particleCalorimeter/param.hpp]: authors added (and renaming)

File: include/picongpu/plugins/transitionRadiation/executeParticleFilter.hpp [hunk 7]
```diff
@@ -47,6 +47,6 @@ namespace picongpu::plugins::transitionRadiation
             = pmacc::traits::HasIdentifier<typename T_Species::FrameType, transitionRadiationMask>::type::value;
 
         if constexpr(hasTransitionRadiationFilter)
-            particles::Manipulate<picongpu::plugins::transitionRadiation::GammaFilter, T_Species>{}(currentStep);
+            particles::manipulate<picongpu::plugins::transitionRadiation::GammaFilter, T_Species>(currentStep);
```
Comment (PrometheusPi) [inline include/picongpu/plugins/transitionRadiation/executeParticleFilter.hpp:50]: When was this renamed to a lower case `manipulate`?

File: include/picongpu/traits/attribute/DampedWeighting.hpp [hunk 8]
```diff
@@ -64,7 +64,7 @@ namespace picongpu
              * @param multiplier weighting multiplier, values of < 1 result in damping, > 1 result in boosting
              */
             template<typename T_Particle>
-            HDINLINE void dampWeighting(T_Particle& particle, float_X const multiplier)
+            HDINLINE void dampWeighting(T_Particle& particle, float_X minWeighting, float_X const multiplier)
```
Comment (PrometheusPi) [inline include/picongpu/traits/attribute/DampedWeighting.hpp:67]: new functionality introduced - is this intened for this PR?

Comment (PrometheusPi) [reply include/picongpu/traits/attribute/DampedWeighting.hpp:67]: Ah, just changed interface to avoid extracting it later - fine

**Review by PrometheusPi — DISMISSED** [review]
There are a few questions that might be just might be issues with git. Would be great if you would comment on them @psychocoderHPC.

Other than that it looks good.

File: include/picongpu/fields/absorber/param.hpp
Comment (psychocoderHPC) [reply include/picongpu/fields/absorber/param.hpp:1]: `extensionUnitless.loader` is removed and github is thinking the new file `param.hpp` is the old file + small chanes

File: include/picongpu/logging.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/logging.hpp:29]: this single line is pulling only a single function from pmacc into the picongpu namespace.
This avoids that I need to write everywhere `pmacc::log`

This is much better than `using namespace pmacc;`

File: include/picongpu/particles/DeviceHeap.hpp
Comment (psychocoderHPC) [reply include/picongpu/particles/DeviceHeap.hpp:1]: This solves cyclic includes. In principle only this small object is used but since it was in particle it always pulled a big bunch of include dependencies.
The change is required else the code will not compile.

File: include/picongpu/particles/InitFunctors.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/InitFunctors.hpp:114]: This code has moved from `Particle.tpp` it was a cyclic include dependency which was only that way solvable.
There is no logic change and the function moved was only called from this place.

File: include/picongpu/particles/Manipulate.def
Comment (psychocoderHPC) [reply include/picongpu/particles/Manipulate.def:1]: yes the signature moved out of `Manipulate.hpp` and I am not allowed to create a file name `manipulate.hpp` because some users using strange OS without case sensitivity ^^

File: include/picongpu/particles/Particles.tpp (hunk 3 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/Particles.tpp:379]: This is moved to another class to resolve include issues.

File: include/picongpu/particles/param.hpp (hunk 4 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/param.hpp]: github is mixing removed and newly created files.

File: include/picongpu/plugins/output/param.hpp (hunk 5 above)
Comment (psychocoderHPC) [reply include/picongpu/plugins/output/param.hpp]: github is mixing removed and newly created files.

File: include/picongpu/plugins/particleCalorimeter/param.hpp (hunk 6 above)
Comment (psychocoderHPC) [reply include/picongpu/plugins/particleCalorimeter/param.hpp]: github is mixing removed and newly created files.

File: include/picongpu/plugins/transitionRadiation/executeParticleFilter.hpp (hunk 7 above)
Comment (psychocoderHPC) [reply include/picongpu/plugins/transitionRadiation/executeParticleFilter.hpp:50]: solves include issues by using signature from `Manipulate.def` and the change is equivalent. The new form is btw. better to read

File: include/picongpu/traits/attribute/DampedWeighting.hpp (hunk 8 above)
Comment (psychocoderHPC) [reply include/picongpu/traits/attribute/DampedWeighting.hpp:67]: yes if I read the value here directly from the param than I need to include all particle param files. If you provide it via signature I decrease include dependencies.

A PR for this change will not gain anything and will avoid merging this for at least 4 days.

File: include/picongpu/particles/Manipulate.def
Comment (PrometheusPi) [reply include/picongpu/particles/Manipulate.def:1]: 🤣

**Review by ikbuibui — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
I will merge it with my admin rights. It passed already the CI but due to a conflict of a removed line and a added line in `defines.hpp` the merge conflict was required to be solved. The conflict does not affect compiling.
